src/dao.cpp

```diff
@@ -54,27 +55,114 @@ namespace hypha
       // Valid claim identified - start process
       // process this claim
       Edge::write(get_self(), get_self(), assignment.getHash(), periodToClaim.value().getHash(), common::CLAIMED);
-      int64_t assignmentApprovedDateSec = assignment.getApprovedTime().sec_since_epoch();
+      //int64_t assignmentApprovedDateSec = assignment.getApprovedTime().sec_since_epoch();
       int64_t periodStartSec = periodToClaim.value().getStartTime().sec_since_epoch();
-      int64_t periodEndSec = periodToClaim.value().next().getEndTime().sec_since_epoch();
+      int64_t periodEndSec = periodToClaim.value().getEndTime().sec_since_epoch();
 
       // Pro-rate the payment if the assignment was created during the period being claimed
-      float first_phase_ratio_calc = 1; // pro-rate based on elapsed % of the first phase
+      float first_phase_ratio_calc = 1.f; // pro-rate based on elapsed % of the first phase
+
+      /** This is no longer necesary since addition of TimeShare
       if (assignmentApprovedDateSec > periodStartSec)
       {
-         int64_t elapsed_sec = periodEndSec - assignmentApprovedDateSec;
-         int64_t period_sec = periodEndSec - periodStartSec;
-         first_phase_ratio_calc = (float)elapsed_sec / (float)period_sec;
+          int64_t elapsed_sec = periodEndSec - assignmentApprovedDateSec;
+          int64_t period_sec = periodEndSec - periodStartSec;
+          first_phase_ratio_calc = (float)elapsed_sec / (float)period_sec;
       }
+      */ 
```

Comment:
If no longer necessary, why we don't remove it?

Reply:
Yep, this will be removed for sure. I think I forgot about it 😅